### PR TITLE
critest: Add test for pid namespace and exec

### PR DESCRIPTION
**Pull request by umohnani8**
Add test for pid namespace = container
Add test for pid namespace = pod
Add test for exec where tty and stdin set to true

Addresses issue https://github.com/kubernetes-incubator/cri-tools/issues/237

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@mrunalp PTAL

File: pkg/validate/security_context.go [hunk 1]
```diff
@@ -184,6 +184,56 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+			By("start container")
+			startContainer(rc, containerID)
+			Eventually(func() runtimeapi.ContainerState {
+				return getContainerStatus(rc, containerID).State
+			}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+
+			By("get nginx container pid")
+			command := []string{"cat", "/proc/1/cmdline"}
+			output := execSyncContainer(rc, containerID, command)
+			Expect(string(output)).To(ContainSubstring("pause"))
```
Comment (mrunalp) [inline pkg/validate/security_context.go]: pause depends upon what container is being used by the runtime. I guess doing a not of "master process" will be more portable across runtimes and different pod infra containers.

File: pkg/validate/streaming.go [hunk 2]
```diff
@@ -74,10 +74,38 @@ var _ = framework.KubeDescribe("Streaming", func() {
...
+				Stderr:      true,
+			}
+			req := createExec(rc, execReq)
 
 			By("check the output of exec")
-			checkExec(rc, req)
+			checkExec(rc, req, "hello\n", false)
+		})
+
+		It("runtime should support exec with -it flags", func() {
```
Comment (mrunalp) [inline pkg/validate/streaming.go]: You should make a different PR for exec tests. Also we should add a test for exec without -it in that PR.

File: pkg/validate/security_context.go (hunk 1 above)
Comment (umohnani8) [reply pkg/validate/security_context.go]: okay, fixed.

File: pkg/validate/streaming.go (hunk 2 above)
Comment (umohnani8) [reply pkg/validate/streaming.go]: Removed the exec test, will open another PR with those tests.

**Comment by Random-Liu** [comment]
@feiskyer What is the difference between these 2 test cases? Shouldn't we test one container can/can't not access pid in another container?

**Comment by Random-Liu** [comment]
I see. One is contain, another is not contain. :)
